src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/ComInterfaceInfo.cs

```diff
@@ -101,7 +105,67 @@ public static DiagnosticOrInterfaceInfo From(INamedTypeSymbol symbol, InterfaceD
                     guid ?? Guid.Empty,
                     interfaceAttributeData.Options,
                     syntax.Identifier.GetLocation()),
-                symbol));
+                symbol);
+
+            // Now that we've validated all of our requirements, we will check for some non-blocking scenarios
+            // and emit diagnostics.
+            ImmutableArray<DiagnosticInfo>.Builder nonFatalDiagnostics = ImmutableArray.CreateBuilder<DiagnosticInfo>();
+
+            if (baseSymbol is not null
```

Comment:
Can we comment what this is doing. It isn't clear from reading it, but I think the point is the base type is not one we have from source or it is from source, but the compilation is not the one we're working on? That last preciate is a bit dense.